File: personalizatio-sdk/src/main/java/com/personalizatio/AbstractParams.java
package com.personalizatio;

import android.text.TextUtils;
import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

abstract class AbstractParams<P extends AbstractParams<P>> {
	protected final JSONObject params = new JSONObject();

	public interface ParamInterface {
		String getValue();
	}

	/**
	 * Вставка строковых параметров
	 */
	public P put(P.ParamInterface param, String value) {
		try {
			params.put(param.getValue(), value);
		} catch(JSONException e) {
			Log.e(SDK.TAG, e.getMessage(), e);
		}
		return (P) this;
	}
	public P put(P.ParamInterface param, int value) {
		return put(param, String.valueOf(value));
	}
	public P put(P.ParamInterface param, boolean value) {
		return put(param, value ? "1" : "0");
	}


	/**
	 * Вставка параметров с массивом
	 */
	public P put(P.ParamInterface param, String[] value) {
		try {
			params.put(param.getValue(), TextUtils.join(",", value));
		} catch(JSONException e) {
			Log.e(SDK.TAG, e.getMessage(), e);
		}
		return (P) this;
	}

	JSONObject build() {
		return params;
	}
}

File: personalizatio-sdk/src/main/java/com/personalizatio/AbstractParams.kt
package com.personalizatio

import android.text.TextUtils

internal abstract class AbstractParams<P : AbstractParams<P?>?> {
    protected val params: JSONObject? = JSONObject()

    interface ParamInterface {
        fun getValue(): String?
    }

    /**
     * Вставка строковых параметров
     */
    fun put(param: ParamInterface?, value: String?): P? {
        try {
            params.put(param.getValue(), value)
        } catch (e: JSONException) {
            Log.e(SDK.TAG, e.getMessage(), e)
        }
        return this as P
    }

    fun put(param: ParamInterface?, value: Int): P? {
        return put(param, String.valueOf(value))
    }

    fun put(param: ParamInterface?, value: Boolean): P? {
        try {
            params.put(param.getValue(), value)
        } catch (e: JSONException) {
            Log.e(SDK.TAG, e.getMessage(), e)
        }
        return this as P
    }

    fun put(param: ParamInterface?, value: JSONObject?): P? {
        try {
            params.put(param.getValue(), value)
        } catch (e: JSONException) {
            Log.e(SDK.TAG, e.getMessage(), e)
        }
        return this as P
    }


    /**
     * Вставка параметров с массивом
     */
    fun put(param: ParamInterface?, value: Array<String?>?): P? {
        try {
            params.put(param.getValue(), TextUtils.join(",", value))
        } catch (e: JSONException) {
            Log.e(SDK.TAG, e.getMessage(), e)
        }
        return this as P
    }

    fun build(): JSONObject? {
        return params
    }
}
